store/src/test/java/in/testpress/store/network/NetworkTestMixin.kt

```diff
@@ -0,0 +1,42 @@
+package `in`.testpress.store.network
+
+import `in`.testpress.network.ErrorHandlingCallAdapterFactory
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import com.google.common.io.ByteStreams
+import okhttp3.mockwebserver.MockWebServer
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+
+open class NetworkTestMixin {
+
+    lateinit var mockWebServer: MockWebServer
+
+    @get:Rule
+    val instantTaskExecutorRule = InstantTaskExecutorRule()
+
+    lateinit var service: ProductService
+
+    @Before
+    fun createService() {
+        mockWebServer = MockWebServer()
+        service = Retrofit.Builder()
+                .baseUrl(mockWebServer.url("/"))
+                .addCallAdapterFactory(ErrorHandlingCallAdapterFactory())
+                .addConverterFactory(GsonConverterFactory.create())
+                .build()
+                .create(ProductService::class.java)
+    }
+
+    @After
+    fun stopService() {
+        mockWebServer.shutdown()
+    }
+
+    fun getProductListFromFile(filename: String): String {
```

Comment:
Class name says that it is Network Test Mixin but this method says something about file. Can this mixin be used by any class ? Because class name seems like generic, but this method says it is for product list.

Reply:
yes its generic